frontend/test/pytest/test_measurements_results.py

```diff
@@ -160,6 +160,48 @@ def expval3(x: float):
         observed = expval3(np.pi)
         assert np.isclose(observed, expected)
 
+    def test_hermitian_shots_expval(self, backend):
+        """Test expval Hermitian observables with shots."""
+
+        @qjit
+        @qml.qnode(qml.device(backend, wires=3, shots=1000))
+        def circuit(x, y):
+            qml.RX(x, wires=0)
+            qml.RX(y, wires=1)
+            qml.CNOT(wires=[0, 1])
+            A = np.array(
+                [[complex(1.0, 0.0), complex(2.0, 0.0)], [complex(2.0, 0.0), complex(1.0, 0.0)]]
+            )
+            return qml.expval(qml.Hermitian(A, wires=2) + qml.PauliX(0) + qml.Hermitian(A, wires=1))
+
+        with pytest.raises(
+            RuntimeError,
+            match="For Hermitian Observables, expval calculation is not supported by shots",
+        ):
+            circuit(np.pi / 4, np.pi / 4)
+
+        qml.operation.disable_new_opmath()
+
+    def test_hermitian_shots_var(self, backend):
+        """Test var Hermitian observables with shots."""
+
+        @qjit
+        @qml.qnode(qml.device(backend, wires=3, shots=1000))
+        def circuit(x, y):
+            qml.RX(x, wires=0)
+            qml.RX(y, wires=1)
+            qml.CNOT(wires=[0, 1])
+            A = np.array(
+                [[complex(1.0, 0.0), complex(2.0, 0.0)], [complex(2.0, 0.0), complex(1.0, 0.0)]]
+            )
+            return qml.var(qml.Hermitian(A, wires=2) + qml.PauliX(0) + qml.Hermitian(A, wires=1))
+
+        with pytest.raises(
+            RuntimeError,
+            match="Hermitian observables do not support applyInPlaceShots method",
+        ):
+            circuit(np.pi / 4, np.pi / 4)
```

Comment:
I'm curious how come tests for expval/var with non `qml.Hermitian` observables aren't added?

Reply:
I wonder if it is worth porting over some of the finite-shot measurement tests from here? https://github.com/PennyLaneAI/pennylane/blob/master/pennylane/devices/tests/test_measurements.py#L198

Reply:
We had tests with other observables and shots > 0 in the frontend (both in `test_measurements_results.py` and `test_pytree_args.py`). To pass those tests, I only had to increase the number of shots :) 

 Thanks for the link. I'll add more tests :+1:

Reply:
@josh146 added a new test file for testing these finite-shots features. [54489e5](https://github.com/PennyLaneAI/catalyst/pull/392/commits/54489e5e25745c2003c81acd7470fd9b2c2799e3) those commented tests are related to the bug I mentioned this morning and @multiphaseCFD is currently working on it.

Reply:
Thanks Ali, this is great! very thorough